projects/etos_suite_runner/src/etos_suite_runner/lib/esr_parameters.py

```diff
@@ -57,50 +59,125 @@ def get_status(self) -> dict:
+                artifact_created = request_artifact_created(
+                    self.etos, artifact_id=os.getenv("ARTIFACT")
+                )
+            else:
+                tercc = EiffelTestExecutionRecipeCollectionCreatedEvent()
+                tercc.rebuild(self.tercc)
+                artifact_created = request_artifact_created(self.etos, tercc=tercc)
             self.etos.config.set("artifact_created", artifact_created)
         return self.etos.config.get("artifact_created")
 
     @property
-    def tercc(self) -> EiffelTestExecutionRecipeCollectionCreatedEvent:
+    def tercc(self) -> Union[list[dict], dict]:
         """Test execution recipe collection created event from environment.
 
         :return: Test execution event.
         """
         if self.etos.config.get("tercc") is None:
-            tercc = EiffelTestExecutionRecipeCollectionCreatedEvent()
-            tercc.rebuild(json.loads(os.getenv("TERCC")))
+            tercc = json.loads(os.getenv("TERCC", "{}"))
             self.etos.config.set("tercc", tercc)
         return self.etos.config.get("tercc")
 
     @property
-    def test_suite(self) -> list[dict]:
+    def test_suite(self) -> list[Suite]:
         """Download and return test batches."""
         with self.lock:
             if self.__test_suite is None:
-                tercc = self.tercc.json
-                batch = tercc.get("data", {}).get("batches")
-                batch_uri = tercc.get("data", {}).get("batchesUri")
-                if batch is not None and batch_uri is not None:
-                    raise ValueError("Only one of 'batches' or 'batchesUri' shall be set")
-                if batch is not None:
-                    self.__test_suite = batch
-                elif batch_uri is not None:
-                    json_header = {"Accept": "application/json"}
-                    response = self.etos.http.get(
-                        batch_uri,
-                        headers=json_header,
-                    )
-                    response.raise_for_status()
-                    self.__test_suite = response.json()
-        return self.__test_suite if self.__test_suite else []
+                tercc = json.loads(os.getenv("TERCC", "{}"))
+                if isinstance(tercc, list):
+                    self.__test_suite = [Suite(**suite) for suite in tercc]
+                else:
+                    test_suite = self._eiffel_test_suite(tercc)
+                    # The dataset is not necessary for the suite runner.
+                    self.__test_suite = [Suite.from_tercc(suite, {}) for suite in test_suite]
+        return self.__test_suite or []
+
+    def _eiffel_test_suite(self, tercc: dict) -> list[dict]:
+        """Eiffel test suite parses an Eiffel TERCC even and returns a list of test suites."""
+        batch = tercc.get("data", {}).get("batches")
+        batch_uri = tercc.get("data", {}).get("batchesUri")
+        if batch is not None and batch_uri is not None:
```

Comment:
if None not in (batch, batch_uri) ?

Reply:
I believe that readability is higher in my version